### ISSUE TITLE
Documentation Generation

### ISSUE THREAD

**Issue by Bryan-Meier** [casact/PCDM#3]
I am new to this project and was trying to see if I could generate the documentation using the make.bat. I have been unsuccessful. I have Sphinx v5.0.1 installed. When I run make.bat singlehtml (or html) I get the following output:

Running Sphinx v5.0.1
WARNING: html_static_path entry '_static' does not exist
building [mo]: targets for 0 po files that are out of date
building [singlehtml]: all documents
updating environment: [new config] 0 added, 0 changed, 0 removed

Sphinx error:
root file C:\xxx\xxx\PCDM\docs\index.rst not found

It appears I am missing an index.rst file but I am not sure where I would get that file. Any thoughts?

**Comment by genedan** [comment]
Hey @Bryan-Meier, as far as I can tell, nothing of substance has been written after the initial sphinx-quickstart command, so I may just re-initialize the docs. I'll see if I can leave enough files this time such that the `make html` command will allow others to rebuild from source.

As far as guidelines go, I plan to have just the source .rst files and the minimum administrative files as part of the repo, but dynamically generated files should be left locally.

**Comment by Bryan-Meier** [comment]
Sounds great @genedan. The minimum admin files would be great. I have never used Sphinx before so just being able to build the docs in any form would greatly help me. Thanks again for your efforts!

**Comment by genedan** [comment]
I've re-initialized the docs. The old .rst must have been lost the last time I reinstalled my OS. Let me know if you can run the command `make html` in the docs folder and if so, I can close this issue.

**Comment by Bryan-Meier** [comment]
Hi @genedan,

I can now run make html and it gens the docs. When I open the index.html there isn't much to see. There are 3 links Index, Module Index, and Search Page. The index link take me to a blank page that says index. The Module Index link tries to open py-modindex.html which doesn't exist. The search page looks like a normal search page but doesn't find anything when searching which is what I would anticipate.

If you have time and want to spend the energy to hunt these down please feel free. No worries if you don't though as the build in Snowflake works and gives us a decent idea of how objects are wired together from the relationship perspective. Thanks again for the time and energy spent on this project!

**Comment by genedan** [comment]
The empty initialization of sphinx-quickstart is pretty much where we're at with the project. I had envisioned making an interactive ER diagram where a user would be able to zoom and click on a specific table to see more information about the columns, data types, relationships, etc., although I currently lack the skill to be able to make something like that, so I've moved on for the time being. 

Your best bet is to look at the OMG PDF which contains a very detailed description of the model:

https://www.omg.org/spec/PC/1.0/PDF

The 13 subject area models in the document correspond to the each Python module in the repo. The PDF is a bit unwieldy with very complex diagrams fit to one page with details on each table to be found on pages much later in the documentation which makes it a bit difficult to move back and forth without losing your place. That's what motivates me to make something interactive, although it's just a vision for the time being.
